lib/services/credentials/credential.dao.js

```diff
@@ -145,20 +145,37 @@ dao.removeCredential = function (id, type) {
   return db.del(buildIdKey(type, id));
 };
 
-dao.removeAllCredentials = function (id) {
+/*
+ * Remove all credentials
+ *
+ * @id {String}
+ * @deleteAll {Boolean} Optional - Whether to delete the key / set that contains all creds for users.
+ */
+dao.removeAllCredentials = function (id, deleteAll) {
   const dbTransaction = db.multi();
   const credentialTypes = Object.keys(config.models.credentials.properties);
-  const awaitAllPromises = [];
-  credentialTypes.forEach((type) => {
+  const awaitAllPromises = credentialTypes.map((type) => {
+    const authKey = buildIdKey(type, id);
+
     if (type === 'key-auth') {
       // id in this call is actually consumerId, so we need to get all referenced keyIds
-      awaitAllPromises.push(db.smembers(buildIdKey(type, id)).then(ids => {
-        ids.map(keyId => {
-          dbTransaction.del(buildIdKey(type, keyId));
+      // This returns a list of all keys the user owns
+      return db.smembers(authKey)
+        .then(ids => {
+          // Delete each
+          const promises = ids.map(keyId => {
```

Comment:
You can write this as a single line

`const promises = ids.map(keyId => dbTransaction.del(buildIdKey(type, keyId))`

Reply:
Cool.